tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php

```diff
@@ -136,21 +135,20 @@ public function _before() {
     $queue->newsletterRenderedBody = $this->queueRenderedNewsletterWithoutTracking;
     $queue->setSubscribers([$subscriber->getId()]);
     $this->sendingTask = $queue->save();
+    $this->newsletterRepository->refresh($newsletter);
+    $this->newsletter = $newsletter;
 
     // create newsletter link associations
-    $newsletterLink1 = NewsletterLink::create();
-    $newsletterLink1->hash = '90e56';
-    $newsletterLink1->url = '[link:newsletter_view_in_browser_url]';
-    $newsletterLink1->newsletterId = $this->newsletter->getId();
-    $newsletterLink1->queueId = $this->sendingTask->id;
-    $newsletterLink1->save();
-
-    $newsletterLink2 = NewsletterLink::create();
-    $newsletterLink2->hash = 'i1893';
-    $newsletterLink2->url = 'http://google.com';
-    $newsletterLink2->newsletterId = $this->newsletter->getId();
-    $newsletterLink2->queueId = $this->sendingTask->id;
-    $newsletterLink2->save();
+
+    $newsletterLink1 = (new \MailPoet\Test\DataFactories\NewsletterLink($newsletter))
```

Comment:
You don't need to create the `NewsletterLink` factory twice. You can create it one time and reuse it. The `with` methods rewrite the previous configuration.
When it is possible, I prefer using the use statement for the fully classified name. What is your opinion on this topic?

Reply:
Also, I would remove the blank line before the link creation.

Reply:
Done!